### PR TITLE
Fix issue #416. Add certain domains to dependency collector's exclude list

### PR DESCRIPTION
**Pull request by yantang-msft**
Add some domains as a temporary workaround. Will probably be configurable when the configuration story is clear.

### CODE REVIEW

File: test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/ApplicationInsightsExtensionsTests.cs [hunk 1]
```diff
@@ -314,7 +314,7 @@ public static void RegistersTelemetryConfigurationFactoryMethodThatPopulatesItWi
                 Assert.Equal(1, modules.Count());
 #endif
 
-                var dependencyModule = services.FirstOrDefault<ServiceDescriptor>(t => t.ImplementationType == typeof(DependencyTrackingTelemetryModule));
+                var dependencyModule = services.FirstOrDefault<ServiceDescriptor>(t => t.ImplementationFactory?.GetMethodInfo().ReturnType == typeof(DependencyTrackingTelemetryModule));
```
Comment (SergeyKanzhelev) [inline test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/ApplicationInsightsExtensionsTests.cs]: can you please add a test to validate that the default list of excluded domains is not empty. I wonder whether we can have a unit test that works with the queue and validates that we didn't break the authentication by our headers. Without exposing Azure queue access key

Comment (yantang-msft) [inline test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/ApplicationInsightsExtensionsTests.cs]: I added the test to make sure the correlation header is not added if the domain is in the exclude list. To validate the authentication is not broken, since the validation happens at server side, we may have to at least provide a fake key to create a storage client. Although the case I added is not storage specific, it somehow helps.

File: test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs [hunk 2]
```diff
@@ -11,12 +11,80 @@ namespace SampleWebAppIntegration.FunctionalTest
...
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.DependencyCollector;
+    using Microsoft.Extensions.DependencyInjection;
 
     public class DependencyTelemetryMvcTests : TelemetryTestsBase
     {
         private const string assemblyName = "MVCFramework45.FunctionalTests";
 
         [Fact]
+        public void OperationIdOfOutgoingRequestIsPropagated()
```
Comment (SergeyKanzhelev) [inline test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs]: this supposed to be tested by Dependency Collector Module itself. The only test needed in this repo is that everything was configured correctly

File: test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs [hunk 3]
```diff
@@ -11,12 +11,80 @@ namespace SampleWebAppIntegration.FunctionalTest
...
+            catch (Exception e)
+            {
+                string data = DebugTelemetryItems(telemetries);
+                throw new Exception(data, e);
+            }
+#endif
+        }
+
+        [Fact]
+        public void OperationIdOfOutgoingRequestIsNotPropagatedIfDomainIsExcluded()
```
Comment (SergeyKanzhelev) [inline test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs]: rename test - it tests that configuration of the module can be updated by customer and this update will be used

File: test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs (hunk 2 above)
Comment (yantang-msft) [inline test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs]: It makes sense to me at first. But to test the E2E cross component correlation functionality, the DependencyCollector and the listener (this project) are both needed, so it's hard to say it's whose job. Since it's just a test case, I'll leave it here.

File: test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs (hunk 3 above)
Comment (yantang-msft) [inline test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs]: Renamed.

File: test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs (hunk 2 above)
Comment (yantang-msft) [inline test/MVCFramework45.FunctionalTests/FunctionalTest/DependencyTelemetryMvcTests.cs]: Moved to CorrelationMVCTests.cs
